### PR TITLE
[MAINTENANCE] Better error message when calling "Fluent" methods from wrong Datasource type

### LINKED ISSUE #7912: 'Datasource' object has no attribute 'add_query_asset'
Hi,

**Describe the bug**
Want to add a sql query to my data source, 

follow the page: https://docs.greatexpectations.io/docs/guides/connecting_to_your_data/fluent/database/how_to_connect_to_sql_data_using_a_query
to add a query but the data source does not have the method.

To reproduce, follow the quick start, create a data source, then try to add the query.

import great_expectations as gx
context = gx.get_context()
datasource = context.get_datasource("imydatasource")

>> <great_expectations.datasource.new_datasource.Datasource object at 0x0000019405844090>

table_asset = datasource.add_query_asset(name="my_asset", query="SELECT passenger_count, total_amount FROM yellow_tripdata")

>> AttributeError: 'Datasource' object has no attribute 'add_query_asset'

**Expected behavior**'
Expect to be able to add the query to the data source.

**Environment (please complete the following information):**
 - Operating System: Windows
 - Great Expectations Version: 0.16.12

### LINKED ISSUE #7921: 'Datasource' object has no attribute 'get_asset'
**Describe the bug**
Attempt to connect to a data asset to create expectations

**To Reproduce**
follow the page: https://docs.greatexpectations.io/docs/guides/connecting_to_your_data/fluent/batch_requests/how_to_request_data_from_a_data_asset
to access the asset but the data source does not have the method.

To reproduce, follow the quick start, create a data source, then try to access the asset.

import great_expectations as gx
context = gx.get_context()
datasource = context.get_datasource("imydatasource")

<great_expectations.datasource.new_datasource.Datasource object at 0x0000019405844090>

AttributeError: 'Datasource' object has no attribute 'get_asset'

Expected behavior'
Expect to be able to access the data asset

Environment (please complete the following information):

Operating System: Windows
Great Expectations Version: 0.16.12

### PR DESCRIPTION
**Pull request by Kilo59**
Our `0.16` docs assume that users are using Fluent datasources but when following the docs with a previously persisted "old" Block Style Datasources they run into confusing `'Datasource' object has no attribute <ATTR_NAME>` errors.
https://github.com/great-expectations/great_expectations/issues/7921
https://github.com/great-expectations/great_expectations/issues/7912

https://docs.greatexpectations.io/docs/guides/connecting_to_your_data/fluent/batch_requests/how_to_request_data_from_a_data_asset/

This change implements `__getattr__` on the "old", non-fluent `Datasource` to throw a more helpful `NotImplementedError` if the attribute being accessed is one of the commonly referenced Fluent Datasource methods.

- [x] Description of PR changes above includes a link to [an existing GitHub issue](https://github.com/great-expectations/great_expectations/issues)
- [x] PR title is prefixed with one of: [BUGFIX], [FEATURE], [DOCS], [MAINTENANCE], [CONTRIB]
- [x] Code is linted

    ```
    black .

    ruff . --fix
    ```
- [x] Appropriate tests and docs have been updated

For more details, see our [Contribution Checklist](https://docs.greatexpectations.io/docs/contributing/contributing_checklist), [Coding style guide](https://docs.greatexpectations.io/docs/contributing/style_guides/code_style), and [Documentation style guide](https://docs.greatexpectations.io/docs/contributing/style_guides/docs_style).

### CODE REVIEW

File: great_expectations/datasource/new_datasource.py [hunk 1]
```diff
@@ -421,6 +423,12 @@ def config(self) -> dict:
         return copy.deepcopy(self._datasource_config)
 
 
+NOT_FLUENT_ERROR_MSG: str = (
+    "Not a 'Fluent' Datasource. Please refer to `0.15` docs https://docs.greatexpectations.io/docs/0.15.50/"
+    "\n or recreate your datasource with our new 'Fluent' API https://docs.greatexpectations.io/docs/guides/connecting_to_your_data/connect_to_data_overview/"
+)
```
Comment (Kilo59) [inline great_expectations/datasource/new_datasource.py]: Could use some guidance around the wording of this error message.

Comment (anthonyburdi) [reply great_expectations/datasource/new_datasource.py]: I think it looks good, but how about this tweak?:
```suggestion
NOT_FLUENT_ERROR_MSG: str = (
    "Not a 'Fluent' Datasource. Please refer to `0.15` docs for info on non 'Fluent' Datasources: https://docs.greatexpectations.io/docs/0.15.50/"
    "\n or recreate your datasource with our new 'Fluent' API: https://docs.greatexpectations.io/docs/guides/connecting_to_your_data/connect_to_data_overview/"
)
```

File: great_expectations/datasource/new_datasource.py [hunk 2]
```diff
@@ -483,3 +491,24 @@ def _init_data_connectors(
                 name=name,
                 config=config,
             )
+
+    def __getattr__(self, attr: str):
```
Comment (anthonyburdi) [inline great_expectations/datasource/new_datasource.py:495]: How about a couple tests: 1. That tries to access an existing attr with no error message 2. That tries to access a fluent attr on a non fluent datasource and we get the correct error message 3. That tries to access an attr that doesn't exist and gets the appropriate error message?

**Review by anthonyburdi — APPROVED** [review]
Looks good, approving but with two suggestions.

File: great_expectations/datasource/new_datasource.py (hunk 2 above)
Comment (Kilo59) [reply great_expectations/datasource/new_datasource.py:495]: Will do.

Comment (Kilo59) [reply great_expectations/datasource/new_datasource.py:495]: Done

**Pull request merged**